py-polars/polars/internals/dataframe/frame.py

```diff
@@ -274,20 +274,20 @@ def __init__(
         elif isinstance(data, dict):
             self._df = dict_to_pydf(data, columns=columns)
 
-        elif _NUMPY_TYPE(data) and isinstance(data, np.ndarray):
-            self._df = numpy_to_pydf(data, columns=columns, orient=orient)
-
-        elif _PYARROW_TYPE(data) and isinstance(data, pa.Table):
-            self._df = arrow_to_pydf(data, columns=columns)
-
         elif isinstance(data, Sequence) and not isinstance(data, str):
             self._df = sequence_to_pydf(
                 data, columns=columns, orient=orient, infer_schema_length=50
```

Comment:
Can you also switch the Sequence and Series instance checks?
Sequence check takes 10 times longer than a plain instance check

Reply:
Dropped the unnecessary `and not isinstance(data, str)` that follows the `Sequence` check too. If somebody genuinely wants to init a frame with a string, who am I to stand in their way... ;)
```python
pl.DataFrame("xyz")
# ┌──────────┐
# │ column_0 │
# │ ---      │
# │ str      │
# ╞══════════╡
# │ x        │
# │ y        │
# │ z        │
# └──────────┘
```